Sources/Class-CurlFetchWeb.php

```diff
@@ -48,7 +48,7 @@ class curl_fetch_web_data
 		CURLOPT_RETURNTRANSFER	=> 1, // Get returned value as a string (don't output it)
 		CURLOPT_HEADER			=> 1, // We need the headers to do our own redirect
 		CURLOPT_FOLLOWLOCATION	=> 0, // Don't follow, we will do it ourselves so safe mode and open_basedir will dig it
-		CURLOPT_USERAGENT		=> 'Mozilla/5.0 (Windows NT 6.1; WOW64; rv:11.0) Gecko Firefox/11.0', // set a normal looking useragent
+		CURLOPT_USERAGENT		=> SMF_USER_AGENT, // set a normal looking useragent
```

Comment:
At first glance, I'm uneasy about changing this line. I suspect that there may have been a good reason to set a user agent that looks like a normal browser in this case.

In theory, the user agent shouldn't cause any difference in how the remote server responds, but in practice it is entirely possible that a badly written UA filter on the far end would cause the server to respond differently to a UA that didn't look like a normal browser.

Further investigation or discussion could convince me to lay aside my concerns, of course. 🙂

Reply:
These days the UA isn't really used to filter out bad guys anymore, the most common thing for filtering out bad bots and stuff like that is by data centers like aws or azure, I know people who ban entire IP ranges of Amazon Web Services

Reply:
I would be worried almost the other way around - not filtering requests, but filtering responses.  This is used in our fetch_web_data routine, which is used in a few places, e.g., the image proxy.  I've seen lots of funky responses - and lack of response - from other websites while I was testing the proxy.   

We may need 'mozilla' and 'gecko', etc., for compatibility.  For interacting with old poorly-configured web servers.  _The same reasons your browser stuffs 'mozilla' and 'gecko' and 'safari' in your browser's user-agent..._  

(Yes, I worry a lot...)

Reply:
We're thinking similarly, @sbulen. It is silly in theory to have to stuff the UA string all that cruft, but silly stuff happens all the time on the Internet. 😑  There's a reason why nearly all spiders and proxies out there do it.

That said, I do like what @MissAllSunday is proposing in this PR. Maybe if we just change the value of the constant to something that looks more like a typical UA but still has the unique bits necessary to identify SMF as SMF, that would give us the best balance. We'd want to check with @jdarwood007 to make sure the scripts on simplemachines.org were ready to accept that UA before merging it in, though.

Reply:
The stats collection scripts do not validate a user agent.

Reply:
What user agent should we change the constant to?

Reply:
This would probably work for all our needs:
```
define('SMF_USER_AGENT', 'user-agent: Mozilla/5.0 (' . php_uname('s') . ' ' . php_uname('m') . ') AppleWebKit/605.1.15 (KHTML, like Gecko) PHP/' . PHP_VERSION . ' SMF/' . strtr(SMF_VERSION, ' ', '.'));
```

Reply:
Oh, one more thing. Here in Class-CurlFetchWeb.php we don't want the constant to include the `user-agent: ` bit at the start, so we should define the constant as just `Mozilla/5.0 (...` and then prepend the `user-agent: ` bit as necessary elsewhere.

Reply:
Did you guys had in mind that this information is send to every page which is called from smf?

Many of this information shouldn't make public (you don't know if the connection is encrypted) and the receiver trust worthy.

Secondly you should test this agent vs many side to get a better "feeling" if this custom ua realy create no new issues.

Reply:
I don't see how this basic info could create a security issue. There are, of course, plenty of people out there with far more knowledge than me on this matter, so I could be wrong. But since none of those people working at Mozilla, Apple, Google, or Microsoft seem to think there is any risk from putting such information in their UA strings, I think we're safe.

Reply:
I suppose we can leave out the PHP version part if that makes everyone more comfortable. It's not like that bit really matters, after all.

Reply:
Yeah these days sending the server name and the PHP version doesn't do much, considering it is SMF the one that is actually sending that info while trying to communicate with others rather than trying to fetch said info from them.

If you worry about someone intercepting the communication and getting the info, they just don't bother doing that, its way more easy and profitable to just smash everyone with everything and wait until something out of it works.

Reply:
That had nothing related with interception,
you changed the code of the proxy too,
so you send this information all over the place,
when ssl image caching is enabled.